### PR TITLE
Issue #795 Add pipeline which periodically updates the projects dependencies to the latest version

### LINKED ISSUE #795: Create Pixi dependencies update pipeline

### PR DESCRIPTION
**Pull request by Manangka**
Fixes #795

This commit adds a pipeline that periodically updates the project dependencies.
It runs every sunday at 15:00. When changes are found a new update branch is pushed and a PR is created.

The created PR in turn will trigger the normal build pipeline so all the regular checks are run against this branch.
If any of the tests fail they should be fixes as part of the created PR

Reviewers are automatically assigned in this case that are @luitjansl and @JoerivanEngelen 

# Checklist

- [x] Links to correct issue
- [ ] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [ ] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

**Comment by Manangka** [comment]
@Hofer-Julian You might find this interesting as well

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -422,4 +424,89 @@ object NightlyTests : BuildType({
...
+
+    steps {
+        powerShell {
+            name = "Download Github CLI"
+            id = "Download_Github_CLI"
+            formatStderrAsError = true
+            scriptMode = script {
+                content = """
+                    Invoke-WebRequest -Uri  https://github.com/cli/cli/releases/download/v2.42.1/gh_2.42.1_windows_amd64.zip -OutFile gh.zip
+                    Expand-Archive gh.zip -DestinationPath gh
```
Comment (Hofer-Julian) [inline .teamcity/settings.kts:457]: You can just add `gh` as pixi dependency.
Alternative it's also available as module on linux called `github`.
I would suggest the former though.

File: .teamcity/settings.kts [hunk 2]
```diff
@@ -422,4 +424,89 @@ object NightlyTests : BuildType({
...
+            name = "Update dependencies"
+            id = "Update_dependencies"
+            workingDir = "imod-python"
+            scriptMode = script {
+                content = """
+                    ${'$'}env:Path = '%system.teamcity.build.checkoutDir%\gh\bin;' + ${'$'}env:Path  
+                    
+                    echo "Create update branch"
+                    
+                    git remote set-url origin https://%GT_USER%:%env.GH_TOKEN%@github.com/Deltares/imod-python.git
```
Comment (Hofer-Julian) [inline .teamcity/settings.kts:471]: Why is that necessary?

Comment (Manangka) [reply .teamcity/settings.kts:471]: The credentials used during checkout aren't strored. This is by design by TeamCity for security purposes.
Therefor we have to reapply the user and token to be able to commit to the branch

File: .teamcity/settings.kts (hunk 1 above)
Comment (Manangka) [reply .teamcity/settings.kts:457]: I'll be going for your first suggestion. Mainly because I don't want to introduce a new scripting language besides the ones that already being used

**Review by luitjansl — APPROVED** [review]
